docs/source/deployment.rst

```diff
@@ -0,0 +1,247 @@
+
+Run postgres server
+"""""""""""""""""""
+
+In order to run postgres server locally, you should check postgres `website <https://www.postgresql.org/docs/current/static/server-start.html>`__ and follow their description. Generally, postgres server runs automatically when the system starts, but this should be checked in the configuration of the system. 
+
+Run redis server
+""""""""""""""""
+
+Redis is known for a simple run process. When it is installed in the system, it enough to execute `redis-server` binary without arguments. Just in case, check their `website <https://redis.io/topics/quickstart>`__.
+
+Run iroha daemon (irohad)
+"""""""""""""""""""""""""
+
+There is a list of preconditions which you should meet before proceeding:
+
+ * Postgres server is up and running
+ * Redis server is up and running
+ * `irohad` Iroha daemon binary is built and accessible in your system
+ * The genesis block and configuration files were created
+ * Config file uses valid postgres and redis connection settings
+ * A keypair for the peer is generated
+ * This is the first time you run the Iroha on this peer and you want to create new chain
+
+.. Hint:: Have you got something that is not the same as in the list of assumptions? Please, refer to the section below the document, titled as `Dealing with troubles`_.
+
+In case of valid assumptions, the only thing that remains is to launch the daemon process with following parameters:
+
++---------------+-----------------------------------------------------------------+
+| Parameter     | Meaning                                                         |
++---------------+-----------------------------------------------------------------+
+| config        | configuration file, containing postgres, and redis connection,  |
+|               | and values to tune the system                                   |
++---------------+-----------------------------------------------------------------+
+| genesis_block | initial block in the ledger                                     |
++---------------+-----------------------------------------------------------------+
+| keypair_name  | private and public key file names without file extension,       |
+|               | used by peer to sign the blocks                                 |
++---------------+-----------------------------------------------------------------+
+
+An example of shell command, running Iroha daemon is 
+
+.. code-block:: shell
+
+    irohad --config example/config.sample --genesis_block example/genesis.block --keypair_name example/node0
+
+.. Attention:: If you have stopped the daemon and want to use existing chain — you should not pass the genesis block parameter.
+
+
+Docker
+------
+
+In order to run Iroha peer as a single instance in Docker, you should pull the image for Iroha first:
+
+.. code-block:: shell
+
+    docker pull hyperledger/Iroha-docker:latest
```

Comment:
I get `invalid reference format: repository name must be lowercase` when I run this command.

Iroha -> iroha